### DISCUSSION TITLE
Back-off restarting failed container

### DISCUSSION THREAD

**Discussion by doviche** [linkerd/linkerd2#4836] [General] [answered]
Hi colleagues, I am using kubernetes/minikube and I've tried to install my application docker image by using a POD configuration yaml file as follow:

apiVersion: v1
kind: Pod
metadata:
  name: employee-service-pod
spec:
  containers:
    - name: employee-service-cont
      image: doviche/employee-service:latest
      imagePullPolicy: IfNotPresent
      command: [ "echo", "SUCCESS" ]
  restartPolicy: Always
  imagePullSecrets:
    - name: employee-service-secret

So far so good but problem starts just when I try to build my pod by executing:
**kubectl create -f employee-service-pod.yaml -n dev-samples**

When I checkout my POD does show the following error:
**Back-off restarting failed container**

Soon after I ran:
**kubectl describe pod employee-service-pod -n dev-samples**
which shows:

Name:         employee-service-pod
Namespace:    dev-samples
Priority:     0
Node:         minikube/192.168.39.126
Start Time:   Tue, 04 Aug 2020 20:30:52 +0200
Labels:       <none>
Annotations:  <none>
Status:       Running
IP:           172.17.0.6
IPs:
  IP:  172.17.0.6
Containers:
  employee-service-cont:
    Container ID:  docker://8e3fc999c2307d8c303bf5620163bb04c8ad8aabd9bce39d8b53bbc27b8aaaba
    Image:         doviche/employee-service:latest
    Image ID:      docker://sha256:91ef212eba3c90727a206b32cc156f2f7929c783b39bda2c112bf9d3a1ea1ba6
    Port:          <none>
    Host Port:     <none>
    Command:
      echo
      SUCCESS
    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Tue, 04 Aug 2020 20:31:08 +0200
      Finished:     Tue, 04 Aug 2020 20:31:08 +0200
    Ready:          False
    Restart Count:  2
    Environment:    <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-cw85g (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  default-token-cw85g:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-cw85g
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason     Age                From               Message
  ----     ------     ----               ----               -------
  Normal   Scheduled  <unknown>          default-scheduler  Successfully assigned dev-samples/employee-service-pod to minikube
  Normal   Pulled     19s (x3 over 35s)  kubelet, minikube  Container image "doviche/employee-service:latest" already present on machine
  Normal   Created    19s (x3 over 34s)  kubelet, minikube  Created container employee-service-cont
  Normal   Started    19s (x3 over 34s)  kubelet, minikube  Started container employee-service-cont
  Warning  BackOff    7s (x4 over 33s)   kubelet, minikube  Back-off restarting failed container

Honestly I have not been able to identify what's causing the Warning and that's why I've decided to share it with you to see if some good eye sees the arcane.

I appreciate any help as I am stuck on this since so long

I am using minikube v1.11.0 on Linuxmint 19.1.

Many thanks in advance guys.

**Comment by ihcsim** [comment]
@doviche Not sure what's going on there. You can try check the logs to see if there are any hints, using `kubectl logs`. Do you have a Linkerd question?

**Comment by doviche** [comment, marked as answer]
Hi ihscim,
thanks for getting to me, fortunately I've managed to fix it. The problem was related to restartPolicy. I had the value set to Always instead of OnFailure.
Setting the restartPolicy to OnFailure resolved the issue.

Thanks a lot

Douglas
